staging/src/k8s.io/pod-security-admission/test/fixtures_restrictedVolumes.go

```diff
@@ -0,0 +1,379 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/pod-security-admission/api"
+)
+
+/*
+TODO: include field paths in reflect-based unit test
+*/
+
+func init() {
+
+	fixtureData_1_0 := fixtureGenerator{
+		expectErrorSubstring: "restricted volume types",
+		generatePass: func(p *corev1.Pod) []*corev1.Pod {
+			return []*corev1.Pod{
+				// pod that has all allowed volume types
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Volumes = []corev1.Volume{
+						{
+							Name: "volume-configmap",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "volume-configmap-test",
+									},
+									Items: []corev1.KeyToPath{
+										{
+											Key:  "hello",
+											Path: "world",
+										},
+									},
+								},
+							},
+						},
+						{
+							Name: "volume-downwardapi",
+							VolumeSource: corev1.VolumeSource{
+								DownwardAPI: &corev1.DownwardAPIVolumeSource{
+									Items: []corev1.DownwardAPIVolumeFile{
+										{
+											Path: "labels",
+											FieldRef: &corev1.ObjectFieldSelector{
+												FieldPath: "metadata.labels",
+											},
+										},
+									},
+								},
+							},
+						},
+						{
+							Name: "volume-emptydir",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+						{
+							Name: "volume-pvc",
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: "test",
+								},
+							},
+						},
+						{
+							Name: "volume-projects",
+							VolumeSource: corev1.VolumeSource{
+								Projected: &corev1.ProjectedVolumeSource{},
+							},
+						},
+						{
+							Name: "volume-secret",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "test",
+									Items: []corev1.KeyToPath{
+										{
+											Key:  "hello",
+											Path: "world",
+										},
+									},
+								},
+							},
+						},
+					}
+				}),
+			}
+		},
+		generateFail: func(p *corev1.Pod) []*corev1.Pod {
+			return []*corev1.Pod{
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Volumes = []corev1.Volume{
+						{
+							Name: "volume-hostpath",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/dev/null",
+								},
+							},
+						},
+					}
+				}),
```

Comment:
move the hostpath failure case to the end... I might dedupe this with the baseline hostPath check as a follow-up, and would like to not ripple diffs across all other test fixtures